### PR TITLE
Add unbound package.

### LINKED ISSUE #952: [Wolfi Package Request]: Unbound
### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

[Unbound](https://nlnetlabs.nl/projects/unbound/about/) is a validating, recursive, caching DNS resolver. It is designed to be fast and lean and incorporates modern features based on open standards.

Source: https://github.com/NLnetLabs/unbound

### which versions of the software should we include?

1.17.1

### Add some justification for why this specific package and versions are important.

A recursive DNS resolver like unbound is perfectly suited to run in distroless containers where security is paramount.

I am the maintainer of an existing [unbound docker image](https://github.com/klutchell/unbound-docker) based on buildroot and scratch but buildroot is overkill for generating distroless containers with one or two packages.

The existing buildroot recipe is here for reference:
https://github.com/buildroot/buildroot/blob/a45f2af7fae09e4daee9d017b02e8744b357cf48/package/unbound/unbound.mk

I am willing to help maintain this package.

### Please check all that apply

- [X] This package has an un-restrictive license
- [X] The package/versions proposed are actively maintained upstream
- [ ] I am interested in adding this package to Wolfi OS myself
- [X] I am willing to help maintain this package

### PR DESCRIPTION
**Pull request by dlorenc**
Fixes: https://github.com/wolfi-dev/os/issues/952

Related:

#### For new package PRs only

- [X] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [X] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [X] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

**Comment by dlorenc** [comment]
Hey @klutchell - I gave this a try! It builds, could you double check the configure options I used?

File: unbound.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/NLnetLabs/unbound
+      expected-commit: 90831af981221bbce1cd7b15055562336760e484
+      tag: release-${{package.version}}
+  - uses: autoconf/configure
+    with:
+    opts: |
+      --with-libevent
```
Comment (klutchell) [inline unbound.yaml:30]:
```suggestion
      --with-libevent
      --with-libexpat
```

**Comment by klutchell** [comment]
That was fast! I'll take a closer look this week but what about adding `--enable-dnycrypt` with `libsodium`?
https://github.com/buildroot/buildroot/blob/a45f2af7fae09e4daee9d017b02e8744b357cf48/package/unbound/unbound.mk#L45-L50

**Comment by klutchell** [comment]
Is there a way I can test what is produced before merging?

**Comment by dlorenc** [comment]
> That was fast! I'll take a closer look this week but what about adding `--enable-dnycrypt` with `libsodium`? https://github.com/buildroot/buildroot/blob/a45f2af7fae09e4daee9d017b02e8744b357cf48/package/unbound/unbound.mk#L45-L50

Got it tacked on and it still builds!

**Comment by dlorenc** [comment]
> Is there a way I can test what is produced before merging?

I can toss this into an image on ghcr somewhere and send it over!

**Comment by dlorenc** [comment]
You can try it at `ghcr.io/dlorenc/unbound`, I only have an arm64 build though.

**Comment by klutchell** [comment]
> You can try it at `ghcr.io/dlorenc/unbound`, I only have an arm64 build though.

I don't seem to have read access to this.

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**